src/Spago/Command/Fetch.purs

```diff
@@ -423,11 +439,13 @@ getPackageDependencies packageName package = case package of
         , indent $ toDoc errLines
         ]
 
-getWorkspacePackageDeps :: WorkspacePackage -> Dependencies
+getWorkspacePackageDeps :: WorkspacePackage -> ByEnv Dependencies
 getWorkspacePackageDeps pkg =
-  if pkg.hasTests then
-    pkg.package.dependencies <> fromMaybe mempty (map _.dependencies pkg.package.test)
-  else pkg.package.dependencies
+  { core: pkg.package.dependencies
+  , test: fromMaybe (Dependencies Map.empty) do
+      guard pkg.hasTests
```

Comment:
I find `guard` to be confusing more often than not, and this is one of such cases. Could we have an explicit `case` here?

Reply:
Sure